src/client/windshaft.js

```diff
@@ -132,6 +130,9 @@ export default class Windshaft {
      *  - When the filter conditions changed and the dataset should be server-filtered.
      */
     _needToInstantiate(MNS, resolution, filtering) {
+        if (!this.metadata) {
+            return true;
```

Comment:
Isn't this redundant? I guess that `this._MNS` will always be undefined the first time, therefore, the next check should return true every time this condition happens, right?